dhall-docs/README.md

```diff
@@ -97,6 +60,377 @@ dhall-docs --input . --output-link ${OTHER_LINK}
+
+To end, if the `|` character isn't on the opening brace, `dhall-docs` will
+ignore the comment without logging on the console. This let you separate your
+own internal block comments from the ones that you want `dhall-docs` to output
+in the final markup.
+
+
+This is an example of an actual file documented using block comments:
+
+```dhall
+{-|
+# My awesome package
+
+This is the header of my package
+
+* a list item in **bold**
+  - a sub item in _italic_
+  - a [link](https://example.com)
+    * a `code` quote
+
+
+A new paragaph
+
+    an indented code block
+-}
+let myAwesomeFunction {- this will be ignored on the documentation -}
+               = \(x : Natural) -> x + 1
+
+in
+{
+  {-|
+  foo
+      bar
+          baz
+  -}
+  myAwesomeFunction
+}
+```
+
+### Line comments
+
+Normal Dhall single-line comments starts with `--` and ends at a newline. For
+example:
+
+```dhall
+-- single line comment
+0
+```
+
+A `dhall-docs` valid single-line comment starts with `--|` and a single whitespace
+character. For example:
+
+```dhall
+--| foo
+```
+
+You can span your documentation in several single-line comments by writing `--` and two
+(2) whitespaces:
+
+```dhall
+--| foo
+--  bar
+--      baz
+```
+
+The `--` prefix defines the indentation for any line. That is, on the above
+example `dhall-docs` will extract the following contents:
+
+```
+foo
+bar
+    baz
+```
+
+If you would like to force an empty line on the documentation output, place an
+empty single line comment in that place. On this example:
+
+```dhall
+--| foo
+--
+--  bar
+```
+
+`dhall-docs` will extract:
+
+```dhall
+foo
+
+bar
+```
+
+The empty `--` line doesn't need the two (2) whitespaces.
+
+If between two single-line comments there is one (1) or more empty lines, the
+comments after those will be ignored. On this example:
+
+
+```dhall
+--| foo
+--  bar
+
+--  baz
+```
+
+will only capture this text:
+
+```
+foo
+bar
+```
+
+Also note that in a set of consecutive single line comments where the `|` is
+not on the first one, the single-line comments before that one will be
+ignored on the rendered documentation. On this example:
+
+```dhall
+--  foo
+--  bar
+--| baz
+--  qux
+```
+
+the following text will be extracted:
+
+```dhall
+baz
+qux
+```
+
+The set of `--` lines needs to be aligned, meaning that a set of lines like this:
+
+```dhall
+--| foo
+  --  bar
+ --  baz
+```
+
+is invalid: `dhall-docs` will ignore this on the generated documentation and
+a warning will be logged on the console.
+
+If the `|` is not after the first `--`, that comment will be ignored by `dhall-docs`
+without logging anything to the console.
+
+If you write a block comment after a set of `dhall-docs` comments, it will be
+ignored, whether it was a valid `dhall-docs` comment or not. On the following
+two examples:
+
+```dhall
+--| foo
+--  bar
+{- qux -}
+```
+
+```dhall
+--| foo
+--  bar
+{-|
+ qux -}
+```
+
+the extracted text will be the following:
+
+```
+foo
+bar
+```
```

Comment:
As mentioned above, I think we should treat subsequent doc comments as errors.

Can we deduplicate these sections BTW?

Reply:
Sure! I'll add another section on the document

Reply:
I merged both sections in a single one and added some examples there. The section is `Mixing the two type of comments`